File: output/llama3.1-8b/avatar/Java/Python/codeforces_67_B.py
import sys

class Codeforces67B:
    def __init__(self):
        self.sc = input()

    def main(self):
        n, k = map(int, self.sc.split())
        bb = list(map(int, self.sc.split()))
        aa = [0] * n
        m = 0
        for a in range(n-1, -1, -1):
            j = 0
            while bb[a] > 0:
                if aa[j] >= a + k:
                    bb[a] -= 1
                j += 1
            for j_ in range(m, j, -1):
                aa[j_] = aa[j_-1]
            aa[j] = a
            m = max(m, j)
        print(*[str(x+1) for x in aa], sep=' ')

if __name__ == "__main__":
    o = Codeforces67B()
    o.main()

File: output/llama3.1-8b/avatar/Python/Java/codeforces_67_B.java
import java.util.Scanner;

public class codeforces_67_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        int n = Integer.parseInt(input.split(" ")[0]);
        int k = Integer.parseInt(input.split(" ")[1]);

        int[] M = new int[n];
        for (int i = 0; i < n; i++) {
            M[i] = Integer.parseInt(scanner.nextLine());
        }

        int m = 0;
        for (int j = 0; j < n; j++) {
            for (int p = 0; p < n; p++) {
                if (M[p] == 0) {
                    System.out.print((p + 1) + " ");
                    break;
                }
            }
            for (int l = Math.max(0, p - k); l <= p; l++) {
                M[l]--;
            }
            M[p]--;
        }
    }
}